code/modules/antagonists/voidwalker/voidwalker_status_effects.dm

```diff
@@ -0,0 +1,77 @@
+/// Allows us to move through glass but not electrified glass. Can also do a little slowdown before passing through
+/datum/status_effect/glass_passer
+	id = "glass_passer"
+	duration = INFINITE
+	alert_type = null
+	/// How long does it take us to move into glass?
+	var/pass_time = 0 SECONDS
+
+/datum/status_effect/glass_passer/on_apply()
+	if(!pass_time)
+		passwindow_on(owner, type)
+	else
+		RegisterSignal(owner, COMSIG_MOVABLE_BUMP, PROC_REF(bumped))
+	owner.generic_canpass = FALSE
+	RegisterSignal(owner, COMSIG_MOVABLE_CAN_PASS_THROUGH, PROC_REF(can_pass_through))
+	return TRUE
+
+/datum/status_effect/glass_passer/on_remove()
+	passwindow_off(owner, type)
+
+/datum/status_effect/glass_passer/proc/can_pass_through(mob/living/carbon/human/human, atom/blocker, direction)
+	SIGNAL_HANDLER
+
+	if(istype(blocker, /obj/structure/grille))
+		var/obj/structure/grille/grille = blocker
+		if(grille.shock(human, 100))
+			return COMSIG_COMPONENT_REFUSE_PASSAGE
```

Comment:
`CanPass` / `CanPassThrough` is not intended to have side effects like this, it's just supposed to be a generic can we / cant we (because it's used in stuff like click code to determine if you can reach something)

The proper place to put this seems to be `COMSIG_MOVABLE_CROSS_OVER`, maybe with a very small change to return a flag to the signal to then return `FALSE` (which will bump the grille instead of allowing the mob to walk through it) (though note, they *will* bump the grille, which means it will attempt to shock them)